src/main/java/it/eng/idsa/dataapp/service/impl/MultiPartMessageServiceImpl.java

```diff
@@ -240,6 +241,62 @@ public HttpEntity createMultipartMessage(String header, String payload/*, String
 		}
 		return multipartEntityBuilder.build();
 	}
+	
+	@Override
+	public HttpEntity createMultipartMessageForm(String header, String payload, String frowardTo, ContentType ctPayload) {
+		MultipartEntityBuilder multipartEntityBuilder = MultipartEntityBuilder.create();
+		multipartEntityBuilder = MultipartEntityBuilder.create().setMode(HttpMultipartMode.STRICT);
+		try {
+			FormBodyPart bodyHeaderPart;
+			ContentBody headerBody = new StringBody(header, ContentType.APPLICATION_JSON);
+			bodyHeaderPart = FormBodyPartBuilder.create("header", headerBody).build();
+			bodyHeaderPart.addField(HTTP.CONTENT_LEN, "" + header.length());
+			multipartEntityBuilder.addPart(bodyHeaderPart);
+
+			FormBodyPart bodyPayloadPart = null;
+			if (payload != null) {
+				ContentBody payloadBody = new StringBody(payload, ctPayload);
+				bodyPayloadPart = FormBodyPartBuilder.create("payload", payloadBody).build();
+				bodyPayloadPart.addField(HTTP.CONTENT_LEN, "" + payload.length());
+				multipartEntityBuilder.addPart(bodyPayloadPart);
+			}
+
+			FormBodyPart headerForwardTo = null;
+			if (frowardTo != null) {
+				ContentBody forwardToBody = new StringBody(frowardTo, ContentType.DEFAULT_TEXT);
+				headerForwardTo = FormBodyPartBuilder.create("forwardTo", forwardToBody).build();
+				headerForwardTo.addField(HTTP.CONTENT_LEN, "" + frowardTo.length());
+				multipartEntityBuilder.addPart(headerForwardTo);
+			}
+
+		} catch (Exception e) {
+			logger.error(e);
+		}
+		return multipartEntityBuilder.build();
+	}
+	
+	@Override
+	public HttpEntity createMultipartMessageMix(String header, String payload, String frowardTo, ContentType ctPayload) {
```

Comment:
Where is this method used in DataApp project?

Reply:
Wanted to make a method similar to createMultipartMessageForm but found out that the entity builder always puts the Content-Transfer-Encoding header which we don't use when sending Mix format. Leftover code, will delete.